ai/coach.py

```diff
@@ -66,13 +71,25 @@ def wait_for_geometry(self):
     def run(self) -> None:
         self.wait_for_geometry()
         self.logger.debug('Running with process ID {}'.format(os.getpid()))
+        if PROFILE_AI:
+            self.pr = cProfile.Profile()
+            self.pr.enable()
+        cycle_tick_count = 0
         try:
             while True:
                 self.main_loop()
                 self.print_frame_rate()
                 sleep(self.cfg['GAME']['ai_timestamp'])
+                cycle_tick_count += 1
+                if cycle_tick_count == PROFILE_DATA_TICK_NUMBER:
```

Comment:
Could be in a self.profiling() function

Reply:
or a decorator around main. we could use the same one in the engine.

Reply:
We cannot make an elegant decorator because calling the dump_stats function too often can cause performance issues.